frame/support/src/inherent.rs

```diff
@@ -36,6 +36,11 @@ pub trait ProvideInherent {
 	const INHERENT_IDENTIFIER: self::InherentIdentifier;
 
 	/// Create an inherent out of the given `InherentData`.
+	///
+	/// NOTE: All checks necessary to ensure that the inherent is correct and that can be done in
+	/// the runtime should happen in the returned `Call`.
```

Comment:
It is not about ensuring that the inherent is correct. If I insert an inherent with me as a block producer, but I'm not allowed to produce a block, the inherent is still correct.

Reply:
I think these docs are not correct here. Maybe we should put them as docs of this module and explain what an inherent is and how to use it?

Reply:
The extrinsic triggered by the inherent should be a correct state transition, no?
So if I want to set the `BlockProducer` storage item in the call returned by `create_inherent` that extrinsic should check that I am allowed to set that value (e.g. by cross-checking with a list of block producers).

:+1: on the second comment. Might make sense to adjust the module level docs.

Reply:
So `sp_inherents` has a pretty good explanation of inherents, so I added a link to that.
I'm still a bit unsure how my comment is wrong.